### DISCUSSION TITLE
Customizing `components` scanning & auto-import

### DISCUSSION THREAD

**Discussion by TechAkayy** [nuxt/nuxt#24169] [Questions]
Following the https://nuxt.com/docs/guide/directory-structure/components page, I tried adding a subfolder to the `components` array in my nuxt config. But, this breaks scanning of components at the base of the "components" folder. `NavBar` component used in my layout, and `DarkModeSwitch` used in the index page are not getting auto-imported anymore. 

Stackblitz - https://stackblitz.com/github/TechAkayy/my-nuxtui-shadcn-vue-x-clone-path-prefix-issue
Github repo - https://github.com/TechAkayy/my-nuxtui-shadcn-vue-x-clone-path-prefix-issue

The only way I was able to make it work, is to add `'~/components'` additionally to this array. 

Question - So, I'm wondering if it's mandatory to include the base components folder to the array anytime we use the `components` key in our nuxt config? If not, this might be an issue. 

I expected the `components` folder to be added to this array by default with `pathPrefix` set to true. 

```
  components: [
    {
      path: '~/components/ui',
      // this is required else Nuxt will autoImport `.ts` file
      extensions: ['.vue'],
      // prefix for your components, eg: UiButton
      prefix: 'My',
      pathPrefix: false,
    },
  ],
```

**Comment by sadeghbarati** [comment]
https://github.com/radix-vue/shadcn-vue/issues/153#issuecomment-1801640488
